components/backup-stream/src/checkpoint_manager.rs

```diff
@@ -0,0 +1,352 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use kvproto::{
+    errorpb::{Error as PbError, *},
+    metapb::Region,
+};
+use pd_client::PdClient;
+use tikv_util::{info, worker::Scheduler};
+use txn_types::TimeStamp;
+
+use crate::{
+    errors::{ContextualResultExt, Error, Result},
+    metadata::{store::MetaStore, CheckpointProvider, MetadataClient},
+    metrics,
+    subscription_track::SubscriptionTracer,
+    try_send, RegionCheckpointOperation, Task,
+};
+
+/// A manager for mataining the last flush ts.
+#[derive(Debug, Default)]
+pub struct CheckpointManager {
+    items: HashMap<u64, LastFlushTsOfRegion>,
+}
+
+#[derive(Debug)]
+pub enum GetCheckpointResult {
+    Ok {
+        region: Region,
+        checkpoint: TimeStamp,
+    },
+    NotFound {
+        id: VersionedRegionId,
+        err: PbError,
+    },
+    EpochNotMatch {
```

Comment:
I think it's helpful to return the region checkpoint too when epoch is not match because br still need to fetch the checkpoint of this region in the retry phase. In addition, if the epoch is change but the region range is not change, br can directly use the checkpoint without retry. This is a tiny optimization, do not affect the correntness of the main logic.